modules/jupyter/main.tf

```diff
@@ -113,24 +113,41 @@ resource "helm_release" "jupyterhub" {
   namespace        = var.namespace
   create_namespace = !contains(data.kubernetes_all_namespaces.allns.namespaces, var.namespace)
   cleanup_on_fail  = "true"
-  # timeout increased to support autopilot scaling resources, and give enough time to complete the deployment 
-  timeout          = 1200
+  # This timeout is sufficient and ensures terraform doesn't hang for 20 minutes on error.
+  # Autopilot deployment will complete even faster than Standard, as it relies on Ray Autoscaler to provision user pods.
+  timeout          = 300
 
-  values = [
-    templatefile("${path.module}/jupyter_config/config-selfauth.yaml", {
+  values = var.autopilot_cluster ? [ templatefile("${path.module}/jupyter_config/config-selfauth-autopilot.yaml", {
       password            = var.add_auth ? "dummy" : random_password.generated_password[0].result
       project_id          = var.project_id
       project_number      = data.google_project.project.number
 
       # Support legacy image.
-      service_id          = ""#var.add_auth ? (data.google_compute_backend_service.jupyter-ingress[0].generated_id != null ? data.google_compute_backend_service.jupyter-ingress[0].generated_id : "no-id-yet") : "no-id-yet"
+      service_id          = "" # TODO(umeshkumhar): var.add_auth ? (data.google_compute_backend_service.jupyter-ingress[0].generated_id != null ? data.google_compute_backend_service.jupyter-ingress[0].generated_id : "no-id-yet") : "no-id-yet"
```

Comment:
describe what the TODO is?

Reply:
I actually don't know what this comment is meant to accomplish. I'm just assigning it for Umesh to investigate. I also left a comment here: https://github.com/GoogleCloudPlatform/ai-on-gke/commit/088d7827902ff541474149ef4ae8ee340d1b74d1#r139178350

Reply:
This will be handled by Liqin PR (https://github.com/GoogleCloudPlatform/ai-on-gke/pull/250).. wil be merged soon.